src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +32,23 @@
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              Todo #{todo.id}
             </div>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button type="button" className="delete" data-cy="modal-close" onClick={onClose} />
```

Comment:
The close button should have accessible text or an `aria-label` to describe its action for screen readers. This ensures that users relying on assistive technologies understand the button's purpose.